packages/govuk-frontend/src/govuk/components/input/template.jsdom.test.js

```diff
@@ -0,0 +1,496 @@
+      expect($component).toHaveAttribute('type', 'number')
+    })
+
+    it('sets the `pattern` attribute based on the `pattern` option', () => {
+      document.body.innerHTML = render(
+        'input',
+        examples['with pattern attribute']
+      )
+
+      const $component = document.querySelector('.govuk-input')
+      expect($component).toHaveAttribute('pattern', '[0-9]*')
+    })
+
+    it('sets the `value` attribute based on the `value` option', () => {
+      document.body.innerHTML = render('input', examples.value)
+
+      const $component = document.querySelector('.govuk-input')
+      expect($component).toHaveValue('QQ 12 34 56 C')
+    })
+
+    it('sets the `value` attribute based on a `value` of "0"', () => {
+      document.body.innerHTML = render('input', examples['zero value'])
+
+      const $component = document.querySelector('.govuk-input')
+      expect($component).toHaveValue('0')
+    })
+
+    it('sets the `aria-describedby` attribute based on the `describedBy` option', () => {
+      document.body.innerHTML = render('input', examples['with describedBy'])
+
+      const $component = document.querySelector('.govuk-input')
+      expect($component).toHaveAttribute(
+        'aria-describedby',
+        'test-target-element'
+      )
+    })
+
+    it('sets any additional attributes based on the `attributes` option', () => {
+      document.body.innerHTML = render('input', examples.attributes)
+
+      const $component = document.querySelector('.govuk-input')
+      expect($component).toHaveAttribute('data-attribute', 'my data value')
+    })
+
+    it('includes additional classes from the `formGroup.classes` option on the form group', () => {
+      document.body.innerHTML = render(
+        'input',
+        examples['with optional form-group classes']
+      )
+
+      const $formGroup = document.querySelector('.govuk-form-group')
+      expect($formGroup).toHaveClass('extra-class')
+    })
+  })
+
+  describe('when a hint is passed', () => {
+    let $component
+    let $hint
+
+    beforeAll(() => {
+      document.body.innerHTML = render('input', examples['with hint text'])
+
+      $component = document.querySelector('.govuk-input')
+      $hint = document.querySelector('.govuk-hint')
+    })
+
+    it('includes the hint', () => {
+      expect($hint).toBeInTheDocument()
+    })
+
+    it('associates the input as described by the hint', () => {
+      expect($component).toHaveAccessibleDescription($hint.textContent.trim())
+    })
+
+    it('associates the input as described by both the hint and the `describedBy` option', () => {
+      document.body.innerHTML = render(
+        'input',
+        examples['hint with describedBy']
+      )
+
+      const $additionalDescription = document.createElement('p')
+      $additionalDescription.id = 'test-target-element'
+      $additionalDescription.textContent = 'Additional description'
+      document.body.appendChild($additionalDescription)
```

Comment:
**question** Is this necessary for `toHaveAccessibleDescription` to function properly (finding the element with the relevant ID and computing its accessible name for use as a description, which it cannot do if the element is not in the DOM)?

Reply:
Yes, if we want to be able to use `toHaveAccessibleDescription` and be confident that the external element referenced by the `describedBy` option is included.

The alternative would be to assert against the value of the `aria-describedby` attribute itself (like the old tests did) but I felt it was better to have just one way of doing it?

Reply:
I agree, it's much nicer to use the 'toHaveAccessibleDescription' rather than look at the `aria-describedby. Just wanted to check I understood why that paragraph was there 😊